File: fuentes/sapiens/src/main/java/com/vortexbird/sapiens/service/PruebaUsuarioService.java
package com.vortexbird.sapiens.service;

import java.util.List;

import com.vortexbird.sapiens.domain.PruebaUsuario;
import com.vortexbird.sapiens.dto.PruebaUsuarioDTO;


/**
* @author Zathura Code Generator http://zathuracode.org/
* www.zathuracode.org
*
*/
public interface PruebaUsuarioService extends GenericService<PruebaUsuario, Integer> {

    void iniciarPrueba(Integer prusId, Long usuario) throws Exception;

    void pausarPrueba(Integer prusId, Long usuario) throws Exception;
    
	void finalizarPrueba(Integer prusId, Long usuario) throws Exception;

	List<PruebaUsuarioDTO> getPruebas(Long usuaId, Long prusId) throws Exception;
}

File: fuentes/sapiens-frontend/src/app/services/prueba-usuario.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { PruebaUsuario } from 'app/domain/prueba-usuario';
import { environment } from 'environments/environment';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PruebaUsuarioService {

  url: string;

  constructor(private httpClient: HttpClient) {
    this.url = environment.URL + 'pruebaUsuario/';
  }

  public getPruebas(usuaId: number, prusId: number): Observable<any> {
    return this.httpClient.get(this.url + 'getPruebas/' + usuaId + '/' + prusId);
  }

  public iniciarPrueba(pruebaUsuario: PruebaUsuario): Observable<any> {
    return this.httpClient.post(this.url + 'iniciarPrueba/', pruebaUsuario);
  }

  public pausarPrueba(pruebaUsuario: PruebaUsuario): Observable<any> {
    return this.httpClient.post(this.url + 'pausarPrueba/', pruebaUsuario);
  }

  public finalizarPrueba(pruebaUsuario: PruebaUsuario): Observable<any> {
    return this.httpClient.post(this.url + 'finalizarPrueba/', pruebaUsuario);
  }
}
